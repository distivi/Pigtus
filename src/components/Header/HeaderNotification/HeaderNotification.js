import React from 'react';
import PropsTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default class HeaderNotification extends React.Component {
  getNotificationCount = () => {
    return this.props.count ?
      <div className="notification-counter">{this.props.count}</div> :
      '';
  }

  openNotification = () => {
    const callback = this.props.openNotification;

    if (callback) {
      callback();
    } else {
      console.warn('No notification callback provided.');
    }
  }

  renderNotification = () => {

  }

  render() {
    const count = this.getNotificationCount();
    return (
        <Link className="header-notification"
          onClick={this.openNotification}
          to={this.props.target || '/help'}>
          {count}
        </Link>
    );
  }
}

HeaderNotification.propTypes = {
  count: PropsTypes.number,
  target: PropsTypes.string,
  openNotification: PropsTypes.func
};
