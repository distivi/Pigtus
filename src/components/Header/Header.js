import React from 'react';
import PropTypes from 'prop-types';
import HeaderNotification from './HeaderNotification/HeaderNotification';

export default class Header extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			text: 'Pigtus'
		};
	}

	render() {
		return (<div className="header">
					<button id="btn-log-out" type="button"
							className="header-action">Log out</button>

					<div className="header-title">
						{this.props.title}
					</div>
					<HeaderNotification count={0}/>
				</div>
		);
	}
}

Header.propTypes = {
	title: PropTypes.string
};

