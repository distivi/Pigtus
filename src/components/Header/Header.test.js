import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import HeaderNotification from './HeaderNotification/HeaderNotification';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { StaticRouter } from 'react-router-dom';

describe('<Header/>', () => {
  it('renders Header without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StaticRouter location='/' context={ {} }><Header isHome={true}/></StaticRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it('Header is consistent with mockup', () => {
    const component = renderer.create(<StaticRouter location='/' context={ {} }><Header isHome={true} title="Mock" notificationCount={1} handleLogout={() => {}}/></StaticRouter>);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Header has title "Mock"', () => {
    const expected = 'Mock';

    const router = shallow(<StaticRouter location='/' context={ {} }><Header isHome={true} title="Mock"/></StaticRouter>);
    const header = router.find(Header).shallow();

    const actual = header.find('.header-title').text();

    expect(actual).toBe(expected);
  });

  it('Header notification counter equal to 2', () => {
    const header = shallow(<Header isHome={true} title="Mock" notificationCount={2}/>);

    expect(header.find('.notification-counter').length).toBe(0);
    expect(header.find(HeaderNotification).shallow().find('.notification-counter').length).toBe(1);
    expect(header.find(HeaderNotification).shallow().find('.notification-counter').text()).toBe('2');
  });

  it('Header handleLogout callback is triggered.', () => {
    const handleLogout = sinon.spy();

    const router = shallow(<StaticRouter location='/' context={ {} }><Header isHome={true} title="Mock" handleLogout={handleLogout}/></StaticRouter>);
    const header = router.find(Header).shallow();

    expect(header.find('.header-action').length).toBe(1);

    const logoutButton = header.find('.header-action');
    logoutButton.simulate('click');

    expect(handleLogout.calledOnce).toBeTruthy();
  });
});
