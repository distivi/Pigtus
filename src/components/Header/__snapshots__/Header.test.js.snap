// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Header is consistent with mockup 1`] = `
<div
  className="header"
>
  <button
    className="header-action"
    id="btn-log-out"
    onClick={[Function]}
    type="button"
  >
    Log out
  </button>
  <div
    className="header-title"
  >
    Mock
  </div>
  <a
    className="header-notification"
    href="/help"
    onClick={[Function]}
  >
    <div
      className="notification-counter"
    >
      1
    </div>
  </a>
</div>
`;
